src/components/CustomEdge.tsx

```diff
@@ -1,200 +1,138 @@
-
-        // const markerEnd = `url(#color=${getCurrentColor()}&type=${MarkerType.ArrowClosed})`;
-
-        const [edgeCenterX, edgeCenterY] = useMemo(
-            () => getEdgeCenter({ sourceX, sourceY, targetX, targetY }),
-            [sourceX, sourceY, targetX, targetY]
-        );
-
-        const buttonSize = 32;
-
-        // Prevent hovered state from getting stuck
-        const hoverTimeout = useDebouncedCallback(() => {
-            setIsHovered(false);
-        }, 7500);
-
-        const [, setHoveredNode] = useHoveredNode;
-
-        return (
-            <g
+const CustomEdge = ({
+    id,
+    source,
+    sourceX,
+    sourceY,
+    targetX,
+    targetY,
+    sourcePosition,
+    targetPosition,
+    style = {},
+    selected,
+}: EdgeProps<EdgeData>) => {
+    const edgePath = useMemo(
+        () =>
+            getBezierPath({
+                sourceX,
+                sourceY,
+                sourcePosition,
+                targetX,
+                targetY,
+                targetPosition,
+            }),
+        [sourceX, sourceY, sourcePosition, targetX, targetY, targetPosition]
+    );
+
+    const { removeEdgeById, nodes, useHoveredNode } = useContext(GlobalContext);
+
+    const parentNode = useMemo(() => nodes.find((n) => source === n.id), [source]);
+
+    const [isHovered, setIsHovered] = useState(false);
+
+    // We dynamically grab this data instead since storing the types makes transitioning harder
+    const accentColor = getNodeAccentColors(parentNode?.data.category);
+    const currentColor = selected ? shadeColor(accentColor, -40) : accentColor;
+
+    const [edgeCenterX, edgeCenterY] = useMemo(
+        () => getEdgeCenter({ sourceX, sourceY, targetX, targetY }),
+        [sourceX, sourceY, targetX, targetY]
+    );
+
+    const buttonSize = 32;
+
+    // Prevent hovered state from getting stuck
+    const hoverTimeout = useDebouncedCallback(() => {
+        setIsHovered(false);
+    }, 7500);
+
+    const [, setHoveredNode] = useHoveredNode;
+
+    return (
+        <g
+            style={{
+                cursor: isHovered ? 'pointer' : 'default',
+            }}
+            onDragEnter={() => setHoveredNode(parentNode?.parentNode)}
+            onMouseEnter={() => setIsHovered(true)}
+            onMouseLeave={() => setIsHovered(false)}
+            onMouseOver={() => hoverTimeout()}
+        >
+            <path
+                className="react-flow__edge-path"
+                d={edgePath}
+                id={id}
                 style={{
+                    ...style,
+                    strokeWidth: isHovered ? '4px' : '2px',
+                    stroke: currentColor,
+                    transitionDuration: '0.15s',
+                    transitionProperty: 'stroke-width, stroke',
+                    transitionTimingFunction: 'ease-in-out',
                     cursor: isHovered ? 'pointer' : 'default',
                 }}
-                onDragEnter={() => {
```

Comment:
Why'd you get rid of this? Did it not actually do anything?

Reply:
I didn't. The diff just can't handle the whitespace changes. It's now `onDragEnter={() => setHoveredNode(parentNode?.parentNode)}` in line 65.

Reply:
Oh I see!